apicurio-registry/pkg/controller/apicurioregistry/kubecl.go

```diff
@@ -0,0 +1,146 @@
+package apicurioregistry
+
+import (
+	"encoding/json"
+	"errors"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net"
+	"os"
+)
+
+type KubeCl struct {
+	client kubernetes.Interface
+	ctx    *Context
+}
+
+// This is provides a way to update resources or otherwise talk to Kubernetes
+func NewKubeCl(ctx *Context) *KubeCl {
+	return &KubeCl{client: newKubeClOrDie(), ctx: ctx}
+}
+
+func newKubeClOrDie() kubernetes.Interface {
+	cfg, err := inClusterConfig()
+	if err != nil {
+		panic(err)
+	}
+	return kubernetes.NewForConfigOrDie(cfg)
+}
+
+func inClusterConfig() (*rest.Config, error) {
```

Comment:
This functionality is already given by the controller manager and you have available at ctx.client field.